### DISCUSSION TITLE
How stateless is Lychee? Can I run multiple instances with shared storage?

### DISCUSSION THREAD

**Discussion by kiancross** [LycheeOrg/Lychee#2853] [Q&A] [answered]
How stateless is Lychee? I am running a Lychee installation with S3, and have multiple instances running, all connected to the same S3 backend.

This isn't causing any problems for users accessing the site, but I wonder if it will cause problems for uploads? Is anybody else horizontally scaling Lychee?

**Comment by ildyria** [comment]
It is stateless as long as the DB is shared between instances. Otherwise you won't be able to see the correct pictures.

**Reply by ildyria** [reply]
But I may be wrong.

**Reply by kiancross** [reply]
Yes, the DB is shared. I wasn't quite sure how things like jobs work, whether they are stored in the database (which would be the best case) or if they were locally stored.

**Reply by ildyria** [reply]
In theory the jobs are stored in DB. AFAIK.

**Reply by kiancross** [reply, marked as answer]
I seem to have it working. Key points for others trying to replicate this are to (1) use object storage so each instance is stateless; (2) obviously connect to the same DB; (3) make sure sessions and queues are stored in the DB; and (4) make sure `APP_KEY` is set to the same value on each instance.
